safe_check.py

```diff
@@ -23,7 +30,18 @@
   try:
     value = safe.safe_check(usercode)
     output += str(value)
-  except Exception,e:
+  except Exception, e:
+    # In virtual_namespace.py, we prepend an encoding declaration to 
+    # each user file we read in.
+    # Now, we need to subtract the number of lines consumed by the 
+    # encoding declaration from the current exception's line number 
+    # so that the line number we output corresponds with the actual 
+    # user code again (see SeattleTestbed/repy_v2#95).
+    try:
+      e.lineno = e.lineno - len(virtual_namespace.ENCODING_DECLARATION.splitlines())
+    except AttributeError:
+      # Some exceptions may not have line numbers.  If so, ignore
+      pass
     output += str(type(e)) + " " + str(e)
```

Comment:
Is this the error message the user gets when a module, or code, fails a safety check?
For example:

```
$ python repy.py restrictions.test ut_errorlineno_codesafety.r2py
Specified repy program is unsafe!
Static-code analysis failed with error: Code failed safety check! Error: ("<class 'exception_hierarchy.CheckNodeException'> (6, 'Import')",)
Terminated
```
The `Error: ...` part of the message seems a bit cryptic.